idunn/places/pj_poi.py

```diff
@@ -273,28 +273,54 @@ def get_phone(self):
             None,
         )
 
-    def get_website(self):
-        if not self.data.website_urls:
-            return None
+    def get_website_url_for_type(self, social_network: UrlType) -> Optional[str]:
```

Comment:
Maybe this could be turned into a `get_website_for_type` method so it can be re-used for url and label?

Reply:
It could, but I think it would move the pain to applying `resolve_url` to the results only when it is not `None`, which would require an extra branching after most calls:

```python
    def get_website(self):
        website = self.get_website_url_for_type(UrlType.EXTERNAL_WEBSITE)

        if not website:
            return None

        return resolve_url(website.website_url)
```

Python's `None` is actually quite painful to work with as there is not much syntactic sugar for common operations (such as optional chaining on JS or the awesome set of [utilities](https://doc.rust-lang.org/std/option/enum.Option.html) and `?` operator in Rust) :cry:

Reply:
I get your point :+1: